### PR TITLE
Fix Paypal Standard issues

### PR DESCRIPTION
**Pull request by jitendrapurohit**

### CODE REVIEW

**Comment by eileenmcnaughton** [comment]
Please don't merge this. I'm happy to review the problem you are trying to fix but this is a reversal of what we are trying to acheive. doTransferPayment is deprecated & billing_mode is deprecated

File: CRM/Core/Payment/Form.php [hunk 1]
```diff
@@ -65,11 +65,15 @@ static public function setPaymentFieldsByProcessor(&$form, $processor, $forceBil
...
-    $form->_paymentFields = array_merge($form->_paymentFields, self::getBillingAddressMetadata($processor, $form->_bltID));
     $form->assign('paymentFields', self::getPaymentFields($processor));
-    self::setBillingAddressFields($form, $processor);
-    // @todo - this may be obsolete - although potentially it could be used to re-order things in the form.
-    $form->billingFieldSets['billing_name_address-group']['fields'] = array();
+    // return without adding billing fields if billing_mode = 4 (@todo - more the ability to set that to the payment processor)
+    // or payment processor is NULL (pay later)
+    if (CRM_Utils_Array::value('billing_mode', $processor) != 4) {
+      $form->_paymentFields = array_merge($form->_paymentFields, self::getBillingAddressMetadata($processor, $form->_bltID));
+      self::setBillingAddressFields($form, $processor);
```
Comment (eileenmcnaughton) [inline CRM/Core/Payment/Form.php]: If you are getting billing_fields in paypal standard then the right classes aren't being overriden on the processor

Comment (jitendrapurohit) [inline CRM/Core/Payment/Form.php]: Thanks for reviewing this from your side. I just wanted to clear some points regarding what I'm getting here -

I don't get `billing_fields` to DISPLAY when I select Paypal Standard, but the `billing_fields` are inserted in `$form->_paymentFields` due to this array_merge happening here which leads to [formRule error message](https://github.com/civicrm/civicrm-core/blob/master/CRM/Contribute/Form/Contribution/Main.php#L905) of required field for all the `billing_fields` even if they are not displayed on the page.

Thanks Again.

Comment (eileenmcnaughton) [inline CRM/Core/Payment/Form.php]: Ah - if I compare getBillingAddressMetadata with getPaymentFieldMetadata it looks like the latter is doing some filtering that the former isn't.

Now I think about it there is a reason for that - sometimes the fields are displayed on a different form (in non-core processors). I think they should both behave the same ie. not actually filter the returned fields, but not set them to required if they are not in the other array. 

I'm actually inclined to say NEVER set them to required here & leave that entirely to the validatePaymentInstrument function - I'm 80% to that point of view

**Comment by eileenmcnaughton** [comment]
closing as this is superceded (& partly resolved

**Pull request closed without merging**